src/BenchmarkDotNet/Engines/CacheClearingStrategies/CacheClearingStrategiesFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using BenchmarkDotNet.Portability;
+
+namespace BenchmarkDotNet.Engines.CacheClearingStrategies
+{
+    internal static class CacheClearingStrategiesFactory
+    {
+        public static ICacheClearingStrategy GetStrategy(CacheClearingStrategy cacheClearingStrategy)
+        {
+            if (!RuntimeInformation.IsWindows())
+            {
+                //All methods use pinvoke.
+                return null;
+            }
+
+            switch (cacheClearingStrategy)
+            {
+                case CacheClearingStrategy.None:
+                    return null;
+                case CacheClearingStrategy.Allocations:
+                    return new AllocationsCacheClearingStrategy();
+                case CacheClearingStrategy.Native:
+                    return new NativeCacheClearingStrategy();
+                default:
+                    throw new NotSupportedException($"Not supported cache clearing strategy: {cacheClearingStrategy}.");
+            }
+        }
+    }
+}
```

Comment:
personally, I don't like the idea of mixing if and switch like that, what do you say about:

```cs
if (!RuntimeInformation.IsWindows() || cacheClearingStrategy == CacheClearingStrategy.None)
    return null;
else if (cacheClearingStrategy == CacheClearingStrategy.Allocations)
    return new AllocationsCacheClearingStrategy();
else if (cacheClearingStrategy == CacheClearingStrategy.Native:
    return new NativeCacheClearingStrategy();
else
    throw new NotSupportedException($"Not supported cache clearing strategy: {cacheClearingStrategy}.");
```

moreover, since you have introduced `ICacheClearingStrategy` you could go fully "object oriented" and instead of returning `null` use "null object pattern"